File: py/src/practise150/P3_two_sum_test.py
from src.practise150.p3_two_sum import Solution

def test_two_sum():
    test_cases =[
        ([3,4,5,6],7,[0,1]),
        ([4,5,6],10,[0,2]),
        ([5,5],10,[0,1])
    ]
    s = Solution()
    for test_case in test_cases:
        result = s.two_sum(test_case[0],test_case[1])
        assert  result == test_case[2]
    for test_case in test_cases:
        result = s.two_sum2(test_case[0],test_case[1])
        assert  result == test_case[2]
    

File: go/practise150/p3_two_sum_test.go
package practise150

import (
	"fmt"
	"slices"
	"testing"
)

func Test_two_sum(t *testing.T) {
	test_cases := []struct {
		input    []int
		target   int
		expected []int
	}{
		{
			input:    []int{3, 4, 5, 6},
			target:   7,
			expected: []int{0, 1},
		},
		{
			input:    []int{4, 5, 6},
			target:   10,
			expected: []int{0, 2},
		},
		{
			input:    []int{5, 5},
			target:   10,
			expected: []int{0, 1},
		},
	}

	for i, test_case := range test_cases {
		t.Run(fmt.Sprintf("Testcase no:%d", i+1), func(t *testing.T) {
			actual := two_sum(test_case.input, test_case.target)
			if !slices.Equal(actual, test_case.expected) {
				t.Errorf("Input: %v, Target: %v, Expected:  %v, Actual: %v", test_case.input, test_case.target, test_case.expected, actual)
			}
		})
	}
}
